src/main/java/org/opentripplanner/middleware/utils/NotificationUtils.java

```diff
@@ -167,75 +168,26 @@ public static boolean sendEmailViaSendGrid(String toEmail, String subject, Strin
-            );
-        }
+    public static boolean isPhoneNumberIsValid(String phoneNumber) {
+        if (StringUtils.isBlank(phoneNumber)) return false;
 
-        com.twilio.rest.lookups.v1.PhoneNumber phoneNumber = null;
-        try {
-            Twilio.init(TWILIO_ACCOUNT_SID, TWILIO_AUTH_TOKEN);
-            phoneNumber = com.twilio.rest.lookups.v1.PhoneNumber.fetcher(
-                new PhoneNumber(phoneNumberString))
-                .setCountryCode(COUNTRY_CODE)
-                .setType(List.of("carrier"))
-                .fetch();
-        } catch (ApiException apiException) {
-            // Handle 404 response - corresponds to invalid number.
-            // In that case we return a 400 bad request response to the requester.
-            if (apiException.getStatusCode() == 404) {
-                logMessageAndHalt(
-                    req,
-                    HttpStatus.BAD_REQUEST_400,
-                    "Phone number format is invalid."
-                );
-            } else {
-                logMessageAndHalt(
-                    req,
-                    HttpStatus.INTERNAL_SERVER_ERROR_500,
-                    "Error validating phone number format."
-                );
-            }
-        } catch (Exception e) {
-            logMessageAndHalt(
-                req,
-                HttpStatus.INTERNAL_SERVER_ERROR_500,
-                "Error validating phone number format."
-            );
-        }
+        // If no regex string/pattern is configured, assume phone number is valid.
+        if (PHONE_PATTERN == null) return true;
 
-        if (phoneNumber != null) {
-            System.out.println(phoneNumber.getPhoneNumber());
-            System.out.println(phoneNumber.getCarrier().get("type"));
-            System.out.println(phoneNumber.getCarrier().get("name"));
-            System.out.println(phoneNumber.getCountryCode());
-            System.out.println(phoneNumber.getNationalFormat());
-
-            // Reject numbers that are international with respect to COUNTRY_CODE.
-            // TODO: Also reject numbers whose .getCarrier().get("type") is not "mobile"?
-            if (!phoneNumber.getCountryCode().equals(COUNTRY_CODE)) {
-                logMessageAndHalt(
-                    req,
-                    HttpStatus.BAD_REQUEST_400,
-                    "Phone number must be domestic."
-                );
-            }
-
-            return phoneNumber;
-        }
+        // Else the phone number must match the regex pattern.
+        Matcher m = PHONE_PATTERN.matcher(phoneNumber);
+        return m.matches();
+    }
 
-        return null;
+    /**
+     * @return A phone number in +155555555 format (E.164 format) for passing to Twilio
+     * by stripping non-digit characters and prefixing with the configured country code.
+     */
+    public static String getRawPhoneNumber(String formattedNumber) {
+        return PHONE_COUNTRY_PREFIX + formattedNumber.replaceAll("\\D", "");
```

Comment:
I think this can go away once we enforce that clients pass phone numbers in E.164.

Reply:
Updated in https://github.com/ibi-group/otp-middleware/pull/83/commits/a9386a6e2f46bc9529e14d51ff9e51106ec87ba6.